lib/reply.js

```diff
@@ -156,16 +156,17 @@ function pumpCallback (reply) {
 
 function handleError (reply, error, cb) {
   var statusCode = reply.res.statusCode
-  if (error == null) {
-    statusCode = statusCode || 500
-  } else if (error.status >= 400) {
-    statusCode = error.status
-  } else if (error.statusCode >= 400) {
-    statusCode = error.statusCode
-  } else {
-    statusCode = statusCode || 500
+
+  if (error) {
```

Comment:
this existential check would be slower in turbofan.

Reply:
Could you describe this in detail? Link?

Reply:
Is it necessary to pre-optimize error paths?

Reply:
Because non-errors get rejected by `handlePromiseError`, will `error` always be truthy here?

Reply:
No chance the error is thrown before.